src/physics/reynolds_averaged_navier_stokes.h

```diff
@@ -0,0 +1,432 @@
+        const std::array<dealii::Tensor<1,dim,real2>,dim+2> &primitive_soln_gradient_rans,
+        const std::array<real2,nstate-(dim+2)> &primitive_soln_turbulence_model) const;
+
+    /// Templated nondimensionalized Reynolds heat flux, (q^reynolds)*
+    template<typename real2> dealii::Tensor<1,dim,real2> compute_Reynolds_heat_flux_templated (
+        const std::array<real2,dim+2> &primitive_soln_rans,
+        const std::array<dealii::Tensor<1,dim,real2>,dim+2> &primitive_soln_gradient_rans,
+        const std::array<real2,nstate-(dim+2)> &primitive_soln_turbulence_model) const;
+
+    template<typename real2> real2 scale_coefficient(
+        const real2 coefficient) const;
+
+private:
+    /// Templated nondimensionalized eddy viscosity for the negative SA model.
+    template<typename real2> real2 compute_eddy_viscosity_templated(
+        const std::array<real2,dim+2> &primitive_soln_rans,
+        const std::array<real2,nstate-(dim+2)> &primitive_soln_turbulence_model) const;
+
+    template<typename real2> std::array<real2,nstate-(dim+2)> compute_effective_viscosity_turbulence_model_templated (
+        const std::array<real2,dim+2> &primitive_soln_rans,
+        const std::array<real2,nstate-(dim+2)> &primitive_soln_turbulence_model) const;
+
+    /// Templated coefficient Chi for the negative SA model.
+    template<typename real2> real2 compute_coefficient_Chi (
+        const real2 &nu_tilde,
+        const real2 &laminar_kinematic_viscosity) const;
+
+    /// Templated coefficient f_v1 for the negative SA model.
+    template<typename real2> real2 compute_coefficient_f_v1 (
+        const real2 &coefficient_Chi) const;
+
+    /// Templated coefficient f_v2 for the negative SA model.
+    real compute_coefficient_f_v2 (
+        const real &coefficient_Chi) const;
+
+    /// Templated coefficient f_n for the negative SA model.
+    template<typename real2> real2 compute_coefficient_f_n (
+        const real2 &nu_tilde,
+        const real2 &laminar_kinematic_viscosity) const;
+
+    /// Templated coefficient f_t2 for the negative SA model.
+    real compute_coefficient_f_t2 (
+        const real &coefficient_Chi) const;
+
+    /// Templated coefficient f_t2 for the negative SA model.
+    real compute_coefficient_f_w (
+        const real &coefficient_g) const;
+
+    /// Templated coefficient r for the negative SA model.
+    real compute_coefficient_r (
+        const real &nu_tilde,
+        const real &d_wall,
+        const real &s_tilde) const;
+
+    /// Templated coefficient g for the negative SA model.
+    real compute_coefficient_g (
+        const real &coefficient_r) const;
+
+    /// Templated vorticity magnitude for the negative SA model.
+    real compute_s (
+        const std::array<real,dim+2> &conservative_soln_rans,
+        const std::array<dealii::Tensor<1,dim,real>,dim+2> &conservative_soln_gradient_rans) const;
+
+    /// Templated correction of vorticity magnitude for the negative SA model.
```

Comment:
I understand that most of those variables names may come from a paper. I would try and avoid using symbols as variable names such as s_bar, Chi, nu, d_wall, f_t2, etc.

I am not knowledgeable enough to know whether **_all_** papers about SA use the same notation, so I will let you judge whether using variable names is reasonable or not. Ideally, someone who's main reference is a paper from another author would be able to understand what exactly is being computer. For example, s_bar = average_vorticity? average_entropy?

Reply:
I will add more comments to detail the variable name and definition in the reference, which follows the code convention of the other class.